lib/sights/index.ts

```diff
@@ -1,107 +1,105 @@
   };
 
-  const updateSight: UpdateSight = (putOptions, callback) => {
-    const requestOptions = {
-      ...optionsToSend,
-      url: `${options.apiUrls.sights}/${putOptions.sightId}`,
-      ...putOptions,
-    };
-    return requestor.put(requestOptions, callback);
+  const copySight = (postOptions: CopySightOptions, callback?: RequestCallback<CopySightResponse>) => {
+    const urlOptions = { url: buildUrl(postOptions.sightId) + '/copy' };
+    return requestor.post({ ...optionsToSend, ...urlOptions, ...postOptions }, callback);
   };
 
-  const copySight: CopySight = (postOptions, callback) => {
-    const requestOptions = {
-      ...optionsToSend,
-      url: `${options.apiUrls.sights}/${postOptions.sightId}/copy`,
-      ...postOptions,
-    };
-    return requestor.post(requestOptions, callback);
+  const moveSight = (postOptions: MoveSightOptions, callback?: RequestCallback<MoveSightResponse>) => {
+    const urlOptions = { url: buildUrl(postOptions.sightId) + '/move' };
+    return requestor.post({ ...optionsToSend, ...urlOptions, ...postOptions }, callback);
   };
 
-  const moveSight: MoveSight = (postOptions, callback) => {
-    const requestOptions = {
-      ...optionsToSend,
-      url: `${options.apiUrls.sights}/${postOptions.sightId}/move`,
-      ...postOptions,
-    };
-    return requestor.post(requestOptions, callback);
+  const getSightPublishStatus = (
+    getOptions: GetSightPublishStatusOptions,
+    callback?: RequestCallback<SightPublishStatus>
+  ) => {
+    const urlOptions = { url: buildUrl(getOptions.sightId) + '/publish' };
+    return requestor.get({ ...optionsToSend, ...urlOptions, ...getOptions }, callback);
   };
 
-  const getSightPublishStatus: GetSightPublishStatus = (getOptions, callback) => {
-    const requestOptions = {
-      ...optionsToSend,
-      url: `${options.apiUrls.sights}/${getOptions.sightId}/publish`,
-      ...getOptions,
-    };
-    return requestor.get(requestOptions, callback);
+  const setSightPublishStatus = (
+    putOptions: SetSightPublishStatusOptions,
+    callback?: RequestCallback<SetSightPublishStatusResponse>
+  ) => {
+    const urlOptions = { url: buildUrl(putOptions.sightId) + '/publish' };
+    return requestor.put({ ...optionsToSend, ...urlOptions, ...putOptions }, callback);
   };
 
-  const setSightPublishStatus: SetSightPublishStatus = (putOptions, callback) => {
-    const requestOptions = {
-      ...optionsToSend,
-      url: `${options.apiUrls.sights}/${putOptions.sightId}/publish`,
-      ...putOptions,
-    };
-    return requestor.put(requestOptions, callback);
+  const buildUrl = (sightId?: number | string) => {
+    if (sightId !== undefined) {
+      return options.apiUrls.sights + '/' + sightId;
+    }
+    return options.apiUrls.sights;
   };
 
   return {
-    listSights,
-    getSight,
-    deleteSight,
-    updateSight,
-    copySight,
-    moveSight,
-    getSightPublishStatus,
-    setSightPublishStatus,
+    getSight: getSight,
```

Comment:
If the key and value are the same, you can use the shortcut:

```suggestion
    getSight,
```

Apply this for the rest and also switch getSight and listSights so that it would appear there is no change to this section of the file.